app/src/main/java/com/andrew/tasky/presentation/agenda/AgendaFragment.kt

```diff
@@ -1,64 +1,279 @@
+                                    .actionAgendaFragmentToLoginFragment()
+                            )
+                            true
+                        }
+                        else -> true
+                    }
+                }
+                popupMenu.show()
+            }
+
+            //shows popup menu allowing user to add different agenda types
+            addAgendaItemFAB.setOnClickListener {
+                val popupMenu = PopupMenu(requireContext(), it)
+                popupMenu.inflate(R.menu.menu_add_agenda_item)
+                popupMenu.setOnMenuItemClickListener {
+                    when (it.itemId) {
+                        R.id.event -> {
+                            val agendaItemType = AgendaItemType.EVENT.name
+                            navController.navigate(
+                                AgendaFragmentDirections
+                                    .actionAgendaFragmentToAgendaItemDetailFragment(agendaItemType)
+                            )
+                            true
+                        }
+                        R.id.reminder -> {
+                            val agendaItemType = AgendaItemType.REMINDER.name
+                            navController.navigate(
+                                AgendaFragmentDirections
+                                    .actionAgendaFragmentToAgendaItemDetailFragment(agendaItemType)
+                            )
+                            true
+                        }
+                        R.id.task -> {
+                            val agendaItemType = AgendaItemType.TASK.name
+                            navController.navigate(
+                                AgendaFragmentDirections
+                                    .actionAgendaFragmentToAgendaItemDetailFragment(agendaItemType)
+                            )
+                            true
+                        }
+                        else -> true
+                    }
+                }
+                popupMenu.show()
+            }
         }
+    }
+
+    private fun setupMiniCalendar(dateSelected: LocalDate) {
+        //miniCalendar is declared in subscribeToObservables function
+        //miniCalendar onClickListeners can be found in onClickListeners function
+
+        // sets up dates of miniCalendar
+        val todayDate = currentDateAndTime.toLocalDate()
+        val tomorrowDate = currentDateAndTime.toLocalDate().plusDays(1)
+        val thirdDayDate = currentDateAndTime.toLocalDate().plusDays(2)
+        val fourthDayDate = currentDateAndTime.toLocalDate().plusDays(3)
+        val fifthDayDate = currentDateAndTime.toLocalDate().plusDays(4)
+        val sixthDayDate = currentDateAndTime.toLocalDate().plusDays(5)
+
+        //sets up dateNumber text view in the mini calendar
+        ItemMiniCalendarDayBinding.bind(fragmentAgendaBinding.today).dateNumber.text =
+            todayDate.format(DateTimeFormatter.ofPattern("d"))
```

Comment:
Instead of re-initializing this `DateTimeFormatter` every time, you could just create it once and use it for all dates